### PR TITLE
PDK Variable Consistency

### LINKED ISSUE #1890: PDK variables to be modified in the flow by OpenLane users
### Description

Most of the PDK variables for `sky130` or `gf180mcu` should not be changed by the users in flow, while some variables are very useful to be modified in the flow. 

### Proposal

Divide the PDK variables into 2 groups, one that the users are encouraged to modify when needed, and the other is specific to the pdk. 
This is a list of the pdk variables that can be modified by the users:
```
FP_PDN_HSPACING
FP_PDN_VSPACING
FP_PDN_HOFFSET
FP_PDN_VOFFSET
FP_PDN_HPITCH
FP_PDN_VPITCH
FP_PDN_HWIDTH
FP_PDN_VWIDTH
FP_PDN_CORE_RING_VWIDTH
FP_PDN_CORE_RING_HWIDTH
FP_PDN_CORE_RING_VSPACING
FP_PDN_CORE_RING_VSPACING
FP_PDN_CORE_RING_VOFFSET
FP_PDN_CORE_RING_HOFFSET
CTS_MAX_CAP
GRT_LAYER_ADJUSTMENTS
RT_MIN_LAYER
RT_MAX_LAYER
FP_IO_VLAYER
FP_IO_HLAYER
FP_TAPCELL_DIST
GPL_CELL_PADDING
DPL_CELL_PADDING
NO_SYNTH_CELL_LIST
DRC_EXCLUDE_CELL_LIST
```
Also, `SYNTH_MAX_TRAN` on top of the list after it is moved to the pdk variables.

### PR DESCRIPTION
**Pull request by donn**
~ Classified PDK variables by user modifiability
~ `SYNTH_MAX_FANOUT` -> Moved into PDK as `MAX_FANOUT_CONSTRAINT`
~ `SYNTH_MAX_TRAN` -> Moved into PDK as `MAX_TRANSITION_CONSTRAINT`
~ `SYNTH_CAP_LOAD` -> `OUTPUT_CAP_LOAD`
- Removed `DEFAULT_MAX_TRAN` from PDK (unused)

---
Blocked by https://github.com/RTimothyEdwards/open_pdks/pull/383
Resolves #1890

### CODE REVIEW

**Comment by kareefardi** [comment]
`SYNTH_MAX_FANOUT` is used in ./regression_results

**Comment by kareefardi** [comment]
I think we might run into errors with SYNTH_MAX_FANOUT and SYNTH_MAX_TRAN with sdc files

**Comment by donn** [comment]
@kareefardi Fixed both- updated the name of the first and used `read_current_sdc` from OL2 for the second.

File: docs/source/reference/pdk_configuration.md [hunk 1]
```diff
@@ -36,37 +63,52 @@ Any examples provided are for the `sky130A` PDK.
...
+
+These values may be modified by the user configuration and the
+included values should be considered "defaults."
+
+| Variable      | Description                                                   |
+|---------------|---------------------------------------------------------------|
+| `MAX_TRANSITION_CONSTRAINT` | Defines the maximum slew (transition) value in ns. |
+| `MAX_FANOUT_CONSTRAINT` | Defines the maximum fanout for a single output in the design. |
+| `OUTPUT_CAP_LOAD` | Defines the capacitive load on the output ports in fF. |
+| `CTS_MAX_CAP` | Defines the maximum capacitance for clock tree synthesis in the design in fF. |
```
Comment (kareefardi) [inline docs/source/reference/pdk_configuration.md:88]: pF according to https://github.com/The-OpenROAD-Project/OpenLane/blob/2b355aaab67eebb8c74b00258b775c710b07b861/scripts/openroad/cts.tcl#L33

**Pull request merged**